lib/rucio/daemons/hermes/kafka/filters/filtered_list.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from rucio.daemons.hermes.kafka.filters.default import default
+from rucio.client import Client
+
+class filtered_list(default):
+
+    RUBIN_BUTLER = "rubin_butler"
+    RUBIN_SIDECAR = "rubin_sidecar"
+
+    def __init__(self, logger, producer, topics):
+        self.logger = logger
+        self.producer = producer
+        self.topics = topics
+        self.client = Client()              
+
+        self.logger(logging.INFO, f'Messages will only be sent to kafka topics: "{self.topics}"')
+
+    def process(self, messages):
+        """Process Hermes events, and send those matching the filter parameters to RSE host locations
+
+        Parameters
+        ----------
+        messages : `list`
+            Hermes events to process
+        """
+        to_delete = []
+        msg_count = 0
+        discard_count = 0
+        # cycle through all the messages, applying the filter rules 
+        # stated above.
+        for message in messages:
+            # if the event_type isn't 'transfer-done', then ignore this message
+            # and mark it for deletion.
+            if str(message['event_type']).lower() != 'transfer-done':
+                discard_count += 1
+                to_delete.append(message['id'])
```

Comment:
I assume each configured messaging system gets to mark messages independently as "to be deleted" and that this is not preventing others from seeing the message.

Reply:
Yes, that's correct.  There are duplicate messages for each of the Rucio-configured systems they serve.  Each has an independent set of messages.